addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1878,7 +1883,16 @@ const SnippetOptionWidget = Widget.extend({
                 if (cssValues.length === 2 && weUtils.areCssValuesEqual(cssValues[1], cssValues[0], params.cssProperty, this.$target)) {
                     cssValues.pop();
                 }
-                return cssValues.join(' ');
+                let value = cssValues.join(' ');
+                if (params.cssProperty === 'box-shadow') {
+                    // matching HEXA | HEX | HEX-shorthand color notations.
+                    const hexColor = value.match(/#([A-Fa-f0-9]{8}|[A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})/g);
+                    const color = hexColor ? hexColor[0] : value.slice(value.indexOf('rgb'), value.indexOf(')') + 1);
+                    const inset = value.indexOf('inset') >= 0 ? 'inset' : '';
+                    value = value.replace(color, '').replace('inset', '');
+                    value = `${color.replace(/\s/g, '')} ${value.trim()} ${inset}`;
+                }
+                return value;
```

Comment:
I know it is technically the same but I would do all of this at the location of the line above `return styles[cssProp].trim();`. In that location, `cssProp` is a basic css property and not a potential shorthand. box-shadow is not part of a shorthand property so it works like this but if more cases like this box-shadow one occur, it may be *required* to do the parsing there.